### ISSUE TITLE
Does not work with Angular 5

### ISSUE THREAD

**Issue by smedasn** [georgipeltekov/ngx-file-drop#17]
ERROR in ./node_modules/ngx-file-drop/lib/ngx-drop/index.ts
Module build failed: Error: TypeScript compilation failed.

**Comment by maartenmensink** [comment]
A pull request for preparing for Angular 5 has already been merged. Perhaps a -beta version can be released? (cc @georgipeltekov)

**Comment by georgipeltekov** [comment]
I reverted the Angular 5 merge, as they were compatibility issues with the older version

**Comment by maartenmensink** [comment]
That is why i added a commit that it might be better to accept the merge an a branch other than the master branch :)

**Comment by georgipeltekov** [comment]
merged. @smedasn could you check with  v1.0.18. Thanks @maartenmensink

**Comment by smedasn** [comment]
Thank you guys. That was really fast. 
Unfortunately I get a warning during "ng serve" now

WARNING in ./node_modules/ngx-file-drop/node_modules/@angular/core/@angular/core.es5.js
5659:15-36 Critical dependency: the request of a dependency is an expression

followed by an error during runtime.

Uncaught (in promise): Error: Unexpected value 'FileDropModule' imported by the module 'ExamplesModule'. Please add a @NgModule annotation.

**Comment by maartenmensink** [comment]
I took a quick stab at debugging the issue but i havent been able to find te cause yet. Will continue next friday.

**Comment by maartenmensink** [comment]
We upgraded our projects to 5.0.0 today and got a different issue. I am wondering if the current npm package format is compatible with the last version of angular.

```
Failed to compile.

./node_modules/ngx-file-drop/lib/ngx-drop/index.ts
Module build failed: Error: C:\projects\butler\UI\Manager\node_modules\ngx-file-drop\lib\ngx-drop\index.ts is not part of the compilation output. Please check the other error messages for details.
    at AngularCompilerPlugin.getCompiledFile (C:\projects\butler\UI\Manager\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:629:23)
    at plugin.done.then (C:\projects\butler\UI\Manager\node_modules\@ngtools\webpack\src\loader.js:467:39)
    at process._tickCallback (internal/process/next_tick.js:109:7)
 @ ./src/app/channel/channel.module.ts 15:0-60
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts
```

**Comment by kevinsalerno** [comment]
This is the second ngx module that is completely broken with Angular 5. Were you guys able to resolve this yet?

Error: Module build failed: Error: ENOENT: no such file or directory, open 'C:\Users\ksalerno\git\identity-data-study-frontend\node_modules\ngx-file-drop\lib\ngx-drop\index.js'
    at Object.<anonymous> (http://localhost:8100/build/vendor.js:119198:7)
    at __webpack_require__ (http://localhost:8100/build/vendor.js:55:30)
    at Object.222 (http://localhost:8100/build/main.js:213:86)
    at __webpack_require__ (http://localhost:8100/build/vendor.js:55:30)
    at Object.198 (http://localhost:8100/build/main.js:188:70)
    at __webpack_require__ (http://localhost:8100/build/vendor.js:55:30)
    at webpackJsonpCallback (http://localhost:8100/build/vendor.js:26:23)
    at http://localhost:8100/build/main.js:1:1

**Comment by maartenmensink** [comment]
@kevinsalerno I got a local build working that publish to a valid angular 4 package format but i have to talk with @georgipeltekov about the build process. I will make a pull-request for the build and publishing process.

**Comment by georgipeltekov** [comment]
fixed in 2.0.0
